plenoptic/synthesize/synthesis.py

```diff
@@ -1075,117 +972,89 @@ def save(self, file_path, save_model_reduced=False, attrs=['model'],
+             check_attributes=['base_signal', 'base_representation',
+                               'loss_function'],
+             objective_function_kwargs={},
+             **pickle_load_args):
+        r"""Load all relevant attributes from a .pt file.
+
+        This should be called by an initialized ``Synthesis`` object -- we will
+        ensure that the attributes in the ``check_attributes`` arg all match in
+        the current and loaded object.
 
-        We will iterate through any additional key word arguments
-        provided and, if the model in the saved representation is a
-        dictionary, add them to the state_dict of the model. In this
-        way, you can replace, e.g., paths that have changed between
-        where you ran the model and where you are now.
+        Note that we check a ``loss_function`` in a special way (because
+        comparing two python callables if very difficult): we compare the
+        outputs on some random images.
+
+        Note this operates in place and so doesn't return anything.
 
         Parameters
         ----------
         file_path : str
             The path to load the synthesis object from
-        model_attr_name : str or list, optional
-            The attribute that gives the model(s) names. Can be a str or
-            a list of strs. If a list and the reduced version of the
-            model was saved, ``model_constructor`` should be a list of
-            the same length.
-        model_constructor : callable, list, or None, optional
-            When saving the synthesis object, we have the option to only
-            save the ``state_dict_reduced`` (in order to save space). If
-            we do that, then we need some way to construct that model
-            again and, not knowing its class or anything, this object
-            doesn't know how. Therefore, a user must pass a constructor
-            for the model that takes in the ``state_dict_reduced``
-            dictionary and returns the initialized model. See the
-            VentralModel class for an example of this. If a list, should
-            be a list of the above and the same length as
-            ``model_attr_name``
         map_location : str, optional
             map_location argument to pass to ``torch.load``. If you save
             stuff that was being run on a GPU and are loading onto a
             CPU, you'll need this to make sure everything lines up
             properly. This should be structured like the str you would
             pass to ``torch.device``
-        state_dict_kwargs :
-            any additional kwargs will be added to the model's
-            state_dict before construction (this only applies if the
-            model is a dict, see above for more description of that)
-
-        Returns
-        -------
-        synthesis : plenoptic.synth.Synthesis
-            The loaded synthesis object
-
+        check_attributes : list, optional
```

Comment:
Regarding this and ``save()``: what if we just force users to save/load very specific attrs. Since we're not tethered to their model anymore and are just dealing w our own Synth objects, we should know/force exactly what is saved and loaded. It could cut down on complexity in the load function.

Reply:
I agree: the user does not set the `check_attribute`: `Synthesis` is not user-facing, it's inherited by other synthesis methods. If you look at `Metamer` and `MADCompetition`, the user doesn't specify the attributes to save/load. But `Metamer` and `MADCompetition` have different attributes to check (in particular, MAD has `base_representation_1` and `base_representation_2`, and handles loss functions slightly differently, so it checks that itself and doesn't have the super class do it); this allows the developer of new synthesis methods to specify which attributes should be checked.

Does that make sense?

Reply:
Ah yes makes sense now. Forgot this was an abc.